src/app/script/palette_class.cpp

```diff
@@ -243,7 +243,7 @@ int Palette_saveAs(lua_State* L)
     if (!ask_access(L, absFn.c_str(), FileAccessMode::Write, ResourceType::File))
       return luaL_error(L, "script doesn't have access to write file %s",
                         absFn.c_str());
-    save_palette(absFn.c_str(), pal, pal->size());
+    save_palette(absFn.c_str(), pal, pal->size(), nullptr);
```

Comment:
If the palette is associated to a sprite (`PaletteObj::spriteId`) we can get the color space from the sprite and use it here.